File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-1.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
       vector<int> x(arr.size());
       x[arr.size()-1]=-1;
       int maxr=-1;
       for(int i=arr.size()-1;i>=0;i--){
        x[i]=maxr;
        maxr=max(arr[i],maxr);
       }
       return x;
    }
};

File: Data Structures & Algorithms/house-robber/submission-1.java
class Solution {
    public int max(int a,int b){
        if(a>b){
            return a;
        }
        return b;
    }
    public int rob(int[] nums) {
             if(nums.length==2){
                return max(nums[0],nums[1]);
             }
             else if(nums.length==1){
                return nums[0];
             }
             else if(nums.length==0){
                return 0;
             }
             int[] dp=new int[nums.length];
             dp[0]=nums[0];
             dp[1]=max(nums[1],nums[0]);
             for(int i=2;i<nums.length;i++){
                dp[i]=max(dp[i-1],dp[i-2]+nums[i]);
             }
            return dp[nums.length-1];
    }
}
